packages/@aws-cdk/cx-api/lib/cloud-artifact.ts

```diff
@@ -80,7 +84,8 @@ export class CloudArtifact {
     switch (artifact.type) {
       case ArtifactType.AWS_CLOUDFORMATION_STACK:
         return new CloudFormationStackArtifact(assembly, id, artifact);
-
+      case ArtifactType.CDK_TREE:
+        return new TreeCloudArtifact(assembly, id, artifact);
       default:
         throw new Error(`unsupported artifact type: ${artifact.type}`);
```

Comment:
**IMPORTANT**: Just realized that this change requires a bump in the cx-protocol version because we were stupid enough to throw an error when reading a cloud assembly with unrecognized artifact types (no future compatibility). This means that an old CLI will crash if it will try to synth a CDK app that has this change.

Please also remove this `default:` section and simply ignore unknown artifacts. That should be the correct behavior going forward so we can avoid such issues.

Reply:
At least make it a warning or something right. Not srue that ignoring is the right behavior.

Reply:
The warning should say something like "Unrecognized artifact type: ${type}. You need to upgrade your CDK CLI to use this artifact" or something. Give users a hint on how to remedy this.

Reply:
I don't think it makes total sense to issue a warning from this library. Maybe the CLI can issue a warning if this returns `undefined`